### PR TITLE
Apply url filters before merging current request's params.

### PR DESCRIPTION
**Pull request by ADmad**
These allows greater control over the url manipulation. For eg. a filter
can prevent the auto setting of "plugin" key based on current request.

### CODE REVIEW

File: src/Routing/Router.php [hunk 1]
```diff
@@ -594,6 +594,8 @@ public static function url($url = null, $full = false)
                 unset($url['_ssl']);
             }
 
+            $url = static::_applyUrlFilters($url);
```
Comment (markstory) [inline src/Routing/Router.php]: This will mean that routes generated with names have very little to no context anymore.

Comment (ADmad) [inline src/Routing/Router.php]: That was the case earlier too. If you see line 599 below only if `_name` was not set the params from current request are merged. So as far as urls with `_name` are concerned this change has no effect.

Comment (markstory) [inline src/Routing/Router.php]: Yes, I misread the code.

Comment (ADmad) [inline src/Routing/Router.php]: One can do the same thing that's done in this method to get context if needed, use `Router::getRequest()`

Comment (markstory) [inline src/Routing/Router.php]: I wonder if this is the kind of behavior change that will catch people off guard when they are updating.

Comment (ADmad) [inline src/Routing/Router.php]: Yes there is a tiny likely hood of that happening though in current form this feature isn't very useful imho. Also I don't think many would be aware of url filters so i would rather change it now than say wait till 3.1

Comment (markstory) [inline src/Routing/Router.php]: Ok.

**Pull request merged**